src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -28,13 +30,17 @@ import { ErrorMessageOutput } from '../error-message-output';
  */
 export abstract class InstructorSessionBasePageComponent {
 
+  // Number of times publish/unpublish session is allowed to error before it is a reportable error
+  private publishUnpublishRetryAttempts: number = 3;
```

Comment:
Maybe we could extract this into a const in the `types` folder. Same with all other retry attempt numbers for more centralised management. Right now it just feels like a magic number to me.